### ISSUE TITLE
map values must be a named type

### ISSUE THREAD

**Issue by tamalsaha** [kubernetes-sigs/controller-tools#287]
My apis package has a Go struct that is not a CRD. But controller-gen keeps throwing the following error:

```console
Generating CRD manifests
/go/src/kubedb.dev/apimachinery/apis/kubedb/v1alpha1/report.go:21:27: map values must be a named type, not *ast.StarExpr
/go/src/kubedb.dev/apimachinery/apis/kubedb/v1alpha1/report.go:22:27: map values must be a named type, not *ast.StarExpr
Error: not all generators ran successfully
```

```go
// +kubebuilder:object:root=false
// +k8s:deepcopy-gen=false
// +k8s:gen-deepcopy=false
type ReportSummary struct {
	Postgres      map[string]*PostgresSummary      `json:"postgres,omitempty"`
	Elasticsearch map[string]*ElasticsearchSummary `json:"elastic,omitempty"`
}
```

Full Example: 
https://github.com/kubedb/apimachinery/blob/report-kubebuilder-bug/apis/kubedb/v1alpha1/report.go

Clone and checkout this branch and then run `make gen` to reproduce.

How do I tell controller-gen to ignore this struct?

**Comment by ifilonenko** [comment]
any progress on this?

**Comment by max8899** [comment]
@tamalsaha try to use the latest controlller-gen in master branch

```
GO111MODULE=on go get -v sigs.k8s.io/controller-tools/cmd/controller-gen@master
```

and re-generate will be work

**Comment by mishindm** [comment]
> @tamalsaha try to use the latest controlller-gen in master branch
> 
> ```
> GO111MODULE=on go get -v sigs.k8s.io/controller-tools/cmd/controller-gen@master
> ```
> 
> and re-generate will be work

Issue is still there:  map values must be a named type, not *ast.InterfaceType

**Comment by jaypipes** [comment]
Yes, this is still an issue. Something where this pops up for me is any API type that contains the [`aws.JSONValue`](https://github.com/aws/aws-sdk-go/blob/776f18cd8fff2e6abd14a0f64a1edc88669ca357/aws/jsonvalue.go#L12) type, which is simply an alias for `map[string]interface{}`.

Until this issue is addressed, I think my only solution is to manually change the CRD field's type from `aws.JSONValue` to `string` and handle the marshal/unmarshal manually within my controller.

Related: https://github.com/aws/aws-controllers-k8s/issues/119

**Comment by PatrickOsborne** [comment]
is this fixed in a particular version of kubebuilder?

**Comment by mrajashree** [comment]
Hi, is this fixed in some version? Still getting this error 
/reopen

**Comment by Callisto13** [comment]
Ditto, using 0.6.0 and still seeing this error.

**Comment by Porges** [comment]
This should be fixed as of #518
